### PR TITLE
update 3.4.3

### PR DESCRIPTION
**Pull request by Arishamays1**
Update `constructor` to 3.4.3

# Package Info

Upstream: https://github.com/conda/constructor
Change log: https://github.com/conda/constructor/blob/main/CHANGELOG.md#2023-02-07---343

# Changes

* Relax `ruamel_yaml` constraints since upstream does not require it: https://github.com/conda/constructor/blob/3.4.3/setup.py#L27
* Add pillow for Unix. `PIL` is imported without any guards, which causes tests to fail: https://github.com/conda/constructor/blob/3.4.3/constructor/imaging.py#L11 This will need to be addressed upstream.

### CODE REVIEW

**Review by marcoesters — DISMISSED** [review]

**Review by ELundby45 — APPROVED** [review]
LGTM

File: recipe/meta.yaml [hunk 1]
```diff
@@ -24,9 +24,9 @@ requirements:
...
     - conda >=4.6,<23.1.0
     - python
-    - ruamel_yaml >=0.11.14,<0.16
+    - ruamel_yaml
     - conda-standalone
-    - pillow >=3.1     # [win or osx]
+    - pillow >=3.1
     - nsis >=3.08      # [win]
   run_constrained:     # [unix]
     - nsis >=3.08      # [unix]
```
Comment (skupr-anaconda) [inline recipe/meta.yaml:32]: `pillow >=3.1` can be removed from `test/requires`, it's already on `run`

Comment (marcoesters) [reply recipe/meta.yaml:32]: Good call, thanks!

**Review by varlackc — APPROVED** [review]
LGTM

**Pull request merged**